06.event_listeners/src/eventListeners.js

```diff
@@ -1,4 +1,57 @@
-var eventListener = (function() {
+const eventListener = (function() {
   'use strict';
-  //code goes here
+  let eventHub = [];
+
+  return {
+    on: (element, eventName, cb) => {
+      if (!eventHub.find(({ el }) => el === element)) {
+        eventHub.push({
+          el: element,
+          events: {},
+        });
+      }
+      const eventHubItem = eventHub.find(({ el }) => el === element);
+
+      if (!(eventName in eventHubItem.events)) {
+        eventHubItem.events[eventName] = [cb];
+        
+        element.addEventListener(eventName, () => {
+          eventHubItem.events[eventName].forEach((callback) => {
+            callback();
+          })
+        })
+      } else {
+        eventHubItem.events[eventName].push(cb);
+      }
+    },
+    off: (element, event, callback) => {
+      if (!element) {
+        eventHub = [];
+        return;
+      }
+      const eventHubItem = eventHub.find(({ el }) => el === element);
+
+      if (!event) {
+        eventHubItem.events = {};
+        return;
+      }
+      if (!callback) {
+        eventHubItem.events[event] = [];
+        return;
+      }
+      eventHubItem.events[event] = eventHubItem.events[event].filter((cb) => {
+        return cb !== callback;
+      });
+    },
+    trigger: (element, eventName) => {
+      const newEvent = new Event(eventName);
+      element.dispatchEvent(newEvent);
+    },
+    delegate: (parentElement, cssName, eventName, callback) => {
+      parentElement.addEventListener(eventName, function (element) {
+        const path = element.composedPath();
+        path.filter(el => el.className === cssName).forEach(() => callback());
```

Comment:
```suggestion
        path
            .filter(el => el.className === cssName)
            .forEach(() => callback());
```
Move this to multiple lines to improve readability